lib/plugin/instance.rb

```diff
@@ -488,6 +489,19 @@ def auth_provider(opts)
     Plugin::AuthProvider.auth_attributes.each do |sym|
       provider.send "#{sym}=", opts.delete(sym)
     end
+
+    begin
+      provider.authenticator.enabled?
+    rescue NotImplementedError
+      # TODO: Remove once plugins are all updated to define enabled? on authenticators
```

Comment:
Hmm this is very hard to enforce since someone in the future looking at the comment wouldn't know which plugins are affected. I think we can just patch it and print the warning.